### PR TITLE
Add row to workspace chat in Workspace

### PR DESCRIPTION
**Pull request by nkdengineer**
### Fixed Issues

$ https://github.com/Expensify/App/issues/51851
PROPOSAL: https://github.com/Expensify/App/issues/51851#issuecomment-2460337283

### Tests

1. Login to the app
2. Go to workspace list
3. Click on a workspace
4. Verify that: `Submit expenses using your workspace chat below:` is displayed in the bottom left
5. Click on workspace in the bottom left
6. Verify that: redirect to workspace chat
- [x] Verify that no errors appear in the JS console

### Offline tests

same as above
### QA Steps

// TODO: These must be filled out, or the issue title must include "[No QA]."

same as above

- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

https://github.com/user-attachments/assets/7599be71-f2ad-427b-b2b6-165f89032604

</details>

<details>
<summary>Android: mWeb Chrome</summary>

https://github.com/user-attachments/assets/7599be71-f2ad-427b-b2b6-165f89032604

</details>

<details>
<summary>iOS: Native</summary>

https://github.com/user-attachments/assets/e6c0cdf6-68e3-49fd-b62f-0e8ba31d2651

</details>

<details>
<summary>iOS: mWeb Safari</summary>

https://github.com/user-attachments/assets/82260aa9-4cc1-4f0f-8a15-595b949c85c2

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/user-attachments/assets/ac169800-e881-47e9-81da-43817928affc

</details>

<details>
<summary>MacOS: Desktop</summary>

https://github.com/user-attachments/assets/910d5712-10c7-43ad-bf26-481b23f9c0eb

</details>

### CODE REVIEW

**Comment by shawnborton** [comment]
Can we use the same exact margin/padding that we have in the Inbox LHN?

Workspace LHN | Inbox LHN
----- | -----
CleanShot 2024-11-13 at 17 15 56@2x | CleanShot 2024-11-13 at 17 15 23@2x

File: src/pages/workspace/WorkspaceInitialPage.tsx
Comment (s77rt) [inline src/pages/workspace/WorkspaceInitialPage.tsx:116]: Instead of getting the report here, we should just get the report id, then use `useOnyx` to subscribe to that report

File: src/pages/workspace/WorkspaceInitialPage.tsx
Comment (s77rt) [inline src/pages/workspace/WorkspaceInitialPage.tsx:463]: The description should be "Workspace" not "Workspaces"

File: src/pages/workspace/WorkspaceInitialPage.tsx
Comment (s77rt) [inline src/pages/workspace/WorkspaceInitialPage.tsx:461]: Would it be better if use `OptionRowLHN` instead?

**Review by s77rt — CHANGES_REQUESTED** [review]

**Comment by s77rt** [comment]
Bug: The offline indicator should be at the bottom

Screenshot 2024-11-13 at 5 47 06 PM

File: src/pages/workspace/WorkspaceInitialPage.tsx
Comment (nkdengineer) [reply src/pages/workspace/WorkspaceInitialPage.tsx:117]: @s77rt To get the report ID, we still need to filter all of reports.

File: src/pages/workspace/WorkspaceInitialPage.tsx
Comment (nkdengineer) [reply src/pages/workspace/WorkspaceInitialPage.tsx:464]: I don't think it should be used `OptionRowLHN` because it shows many other things.

File: src/pages/workspace/WorkspaceInitialPage.tsx
Comment (s77rt) [reply src/pages/workspace/WorkspaceInitialPage.tsx:117]: That's okay but let's use a util function to get the id. We don't want to subscribe to the whole reports collection

**Comment by nkdengineer** [comment]
> Bug: The offline indicator should be at the bottom

@s77rt Fixed this bug.

File: src/pages/workspace/WorkspaceInitialPage.tsx
Comment (s77rt) [inline src/pages/workspace/WorkspaceInitialPage.tsx:98]: There is another `useCurrentUserPersonalDetails` hook, let's remove it

File: src/languages/en.ts
Comment (s77rt) [inline src/languages/en.ts:2473]:
```suggestion
            submitExpense: 'Submit expenses using your workspace chat below:',
```

**Review by s77rt — CHANGES_REQUESTED** [review]

**Comment by dannymcclain** [comment]
@Expensify/design do we want this label to be aligned with the workspace avatar?

CleanShot 2024-11-19 at 07 58 42@2x

Looks like right now it's aligned with the icons from the menu items.

CleanShot 2024-11-19 at 08 00 06@2x

**Comment by nkdengineer** [comment]
> Can we use the same exact margin/padding that we have in the Inbox LHN?

@shawnborton What do you think? This is the reason why I added another margin/padding initially.

**Comment by nkdengineer** [comment]
@s77rt Should we also grey out the submit expense text?

File: src/pages/workspace/WorkspaceInitialPage.tsx
Comment (nkdengineer) [reply src/pages/workspace/WorkspaceInitialPage.tsx]: I updated.

File: src/languages/en.ts
Comment (nkdengineer) [reply src/languages/en.ts]: same

File: src/styles/index.ts
Comment (nkdengineer) [reply src/styles/index.ts]: same

**Comment by s77rt** [comment]
Overall looks good to me. I will retest and approve the PR after the design concerns are addressed.

> Should we also grey out the submit expense text?

I don't have a strong preference, will let this to the design team as well

**Comment by s77rt** [comment]
There is just this "bug?" that I noticed but it exists in the app already (e.g. deep link from chat): If you put the workspace switcher on workspace A and go to workspace B and click on the row. The chat will be open without changing the workspace.


https://github.com/user-attachments/assets/2b4006f8-49a9-4f64-8d21-9abb3fa6ea79

**Comment by shawnborton** [comment]
I would think that label should be aligned with the workspace avatar right? That's what our mock has at least:
image

**Comment by shawnborton** [comment]
> Should we also grey out the submit expense text?

For offline, I don't think we should "gray out" the text above the interactive element, we should only "gray out" the interactive element itself.

**Comment by dannymcclain** [comment]
Agree with both of your comments Shawn!

**Comment by nkdengineer** [comment]
@s77rt Updated the padding of expense row based on the confirmation above.

**Review by s77rt — APPROVED** [review]

**Comment by shawnborton** [comment]
Looks like the proper hover style isn't applied to the avatar, so we don't get the border around the small user avatar to change to the hover color of the row:
CleanShot 2024-11-21 at 09 16 05@2x

**Comment by shawnborton** [comment]
The workspace row should be in the same container as the nav rows above it, we shouldn't have a separate scrollable section: 

https://github.com/user-attachments/assets/c36d4a35-0e2a-43b6-82c0-37a556b70a72

**Comment by shawnborton** [comment]
The padding of the workspace row still isn't correct. Since the outer wrapper here has 12px padding:
CleanShot 2024-11-21 at 09 18 08@2x

We should make the horizontal padding of the workspace row use 8px. This way we get 12px + 8px = 20px total. We need to update the text label above too to reflect that. 

Ideally we should have this:
CleanShot 2024-11-21 at 09 19 37@2x

and on hover:
CleanShot 2024-11-21 at 09 19 49@2x

**Comment by nkdengineer** [comment]
@shawnborton Updated the latest result

- Normal

Screenshot 2024-11-22 at 14 46 20

- Hover

Screenshot 2024-11-22 at 14 46 17

**Comment by shawnborton** [comment]
Feels better! I think we should also reduce the margin between the label and the row, maybe drop it down to just 4px of margin or something?

CleanShot 2024-11-22 at 06 51 20@2x

**Review by grgia — COMMENTED** [review]
Could you also merge main @nkdengineer?

File: src/components/MenuItem.tsx
Comment (nkdengineer) [reply src/components/MenuItem.tsx]: @grgia Merged main and updated.

**Comment by nkdengineer** [comment]
> Feels better! I think we should also reduce the margin between the label and the row, maybe drop it down to just 4px of margin or something?

Also updated the marigin.

**Comment by dannymcclain** [comment]
I think this is looking better, but I'd like Shawn to confirm. As far as I can tell it looks similar to the Inbox presentation.

**Comment by shawnborton** [comment]
I'm still seeing 12px of margin on top of the row instead of 4px:
CleanShot 2024-11-22 at 12 14 12@2x

Otherwise it looks good to me!

**Comment by dannymcclain** [comment]
Ahh it looks like the border on hover is still not quite there.

CleanShot 2024-11-22 at 11 12 51@2x

**Comment by nkdengineer** [comment]
> I'm still seeing 12px of margin on top of the row instead of 4px:

@shawnborton I updated.

**Comment by nkdengineer** [comment]
> Ahh it looks like the border on hover is still not quite there.

@dannymcclain What is the expect here? I see the current hover style is the same as LHN item.

**Comment by shawnborton** [comment]
The border color around the avatar should match the row's background color. They aren't the same right now but they should be.

**Comment by dannymcclain** [comment]
🤔 I'm seeing them as very slightly different. @shawnborton are you seeing something different?

image

Also I noticed some of my workspaces don't show the avatar at all, is that a known thing?

| Header | Header | Header |
|--------|--------|--------|
| CleanShot 2024-11-22 at 11 30 22@2x | CleanShot 2024-11-22 at 11 30 36@2x | CleanShot 2024-11-22 at 11 30 53@2x |

**Comment by shawnborton** [comment]
That looks better! 

@s77rt @grgia let's get this one into final review 👍

File: src/pages/workspace/WorkspaceInitialPage.tsx
Comment (s77rt) [inline src/pages/workspace/WorkspaceInitialPage.tsx:420]:
```suggestion
```
Not needed

File: src/pages/workspace/WorkspaceInitialPage.tsx
Comment (s77rt) [inline src/pages/workspace/WorkspaceInitialPage.tsx:396]:
```suggestion
```
No longer required (related to the above comment ^)

**Review by s77rt — CHANGES_REQUESTED** [review]

**Comment by s77rt** [comment]
The press state has a ring (the background color of the subscript does not match)

Screenshot 2024-11-22 at 7 32 44 PM

**Comment by dannymcclain** [comment]
>The press state has a ring (the background color of the subscript does not match)

Nice catch!

**Comment by nkdengineer** [comment]
@s77rt Removed the unused style and fixed the bug above.

File: src/components/MenuItem.tsx
Comment (s77rt) [inline src/components/MenuItem.tsx:647]: Let's fix this instead of silencing the lint

Comment (nkdengineer) [reply src/components/MenuItem.tsx]: @s77rt Fixed.

**Review by grgia — DISMISSED** [review]
LGTM, let's have design do final confirmation once tests pass

File: src/components/MenuItem.tsx
Comment (s77rt) [inline src/components/MenuItem.tsx:571]:
```suggestion
    const getSubscriptpAvatarBackgroundColor = (isHovered: boolean, isPressed: boolean, hoveredBackgroundColor: string, pressedBackgroundColor: string) => {
```
NAB

File: src/components/MenuItem.tsx
Comment (s77rt) [inline src/components/MenuItem.tsx:577]: NAB. Since we are passing the colors as parameters we should move this function outside the component

**Review by s77rt — APPROVED** [review]

**Comment by nkdengineer** [comment]
> LGTM, let's have design do final confirmation once tests pass

@shawnborton Please help to get a final confirmation when you have a chance, thanks.

**Comment by shawnborton** [comment]
Design looks good to me 👍

**Review by shawnborton — DISMISSED** [review]

**Comment by nkdengineer** [comment]
@grgia Please help to review again when you have a chance.

**Review by grgia — APPROVED** [review]
Nice work on this!

**Review by shawnborton — APPROVED** [review]

**Comment by shawnborton** [comment]
Lovely, feel free to merge @grgia - all good on my end!

File: src/pages/workspace/WorkspaceInitialPage.tsx
Comment (nkdengineer) [reply src/pages/workspace/WorkspaceInitialPage.tsx:449]: @mountiny This case will happen if `isPolicyExpenseChatEnabled` is `false` right?

**Pull request merged**
